plugins/rapid7_insightvm/komand_rapid7_insightvm/util/update_shared_credential_util.py

```diff
@@ -0,0 +1,117 @@
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+from komand_rapid7_insightvm.util.util import check_in_enum, check_not_null
+
+
+def ssh_setup(account: dict):
+    permission_elevation = account.get("permission_elevation", "none")
+    check_in_enum(permission_elevation, "permission_elevation",
+                  ["none", "sudo", "sudosu", "su", "pbrun", "privileged-exec"])
+    if permission_elevation not in ("none", "pbrun"):
+        permission_elevation_username = check_not_null(account, "permission_elevation_username")
+        permission_elevation_password = check_not_null(account, "permission_elevation_password")
+    else:
+        permission_elevation_username = ""
+        permission_elevation_password = ""
+    return permission_elevation, permission_elevation_username, permission_elevation_password
+
+
+def usr_and_pass(account: dict):
+    return check_not_null(account, "username"), check_not_null(account, "password")
+
+
+def as400_cifs_cvs(account: dict, service: str) -> dict:
+    domain = account.get("domain", "")
+    username, password = usr_and_pass(account)
+    return {"service": service, "domain": domain, "username": username, "password": password}
+
+
+def cifs_hash(account: dict, service: str) -> dict:
+    domain = account.get("domain", "")
+    username = check_not_null(account, "username")
+    ntlm_hash = check_not_null(account, "ntlm_hash")
+    return {"service": service, "domain": domain, "username": username, "ntlmHash": ntlm_hash}
+
+
+def db2_mysql_postgresql(account: dict, service: str) -> dict:
+    database = account.get("database", "")
+    username, password = usr_and_pass(account)
+    return {"service": service, "database": database, "username": username, "password": password}
+
+
+def ftp_pop_remote_exec_telnet(account: dict, service: str) -> dict:
+    username, password = usr_and_pass(account)
+    return {"service": service, "username": username, "password": password}
+
+
+def http(account: dict, service: str) -> dict:
+    realm = account.get("realm", "")
+    username, password = usr_and_pass(account)
+    return {"service": service, "realm": realm, "username": username, "password": password}
+
+
+def ms_sql_sybase(account: dict, service: str) -> dict:
```

Comment:
I think this is considered one or the other. Union is _one or the other_
```suggestion
def ms_sql_sybase(account: dict, service: str) -> Union[dict, dict]:
```